### PR TITLE
[DS-4300] Optionally apply given Handles when importing community/collection structure

### LINKED ISSUE #7640: [DS-4300] Optionally apply given Handles when importing community/collection structure
<i>Imported from JIRA <a href="https://jira.lyrasis.org/browse/DS-4300">[DS-4300]</a> created by mwood</i>
<p>Based on <a href="https://github.com/mwoodiupui/DSpace/pull/13" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/mwoodiupui/DSpace/pull/13</a> by <a href="https://jira.lyrasis.org/secure/ViewProfile.jspa?name=pbecker" class="user-hover" rel="pbecker">Pascal-Nicolas Becker</a>.  When importing a community/collection structure that was created by exporting existing structure, the original Handles are available.  This patch adds an option to apply those Handles instead of generating new ones.</p>

### PR DESCRIPTION
**Pull request by mwoodiupui**
Fixes #7640
Based on https://github.com/mwoodiupui/DSpace/pull/13

### CODE REVIEW

**Comment by mwoodiupui** [comment]
Close/open to get Travis to run.

**Comment by tdonohue** [comment]
@mwoodiupui : Could we add tests for this PR in the existing StructBuilderIT? https://github.com/DSpace/DSpace/blob/main/dspace-api/src/test/java/org/dspace/administer/StructBuilderIT.java

File: dspace-api/src/test/java/org/dspace/administer/StructBuilderIT.java
Comment (abollini) [inline dspace-api/src/test/java/org/dspace/administer/StructBuilderIT.java:181]: please remove comment out code

File: dspace-api/src/test/java/org/dspace/administer/StructBuilderIT.java
Comment (abollini) [inline dspace-api/src/test/java/org/dspace/administer/StructBuilderIT.java:232]: is there a reason to apply a different test strategy than the one used in `testImportStructure` via the `DiffBuilder`?
checking for just a single handle could hide lot of (future) issues (maybe the import skip all the community without a handle or handle are not set for collection, etc.)

**Review by abollini — CHANGES_REQUESTED** [review]
hi @mwoodiupui  thanks for this PR and to ping us on the recent bug report about it!
I made a quick code review and added inline a question for you about the test, can you take a look?
my colleague will make a functional test in the next days as well

File: dspace-api/src/test/java/org/dspace/administer/StructBuilderIT.java
Comment (mwoodiupui) [reply dspace-api/src/test/java/org/dspace/administer/StructBuilderIT.java]: Done.

File: dspace-api/src/test/java/org/dspace/administer/StructBuilderIT.java
Comment (mwoodiupui) [reply dspace-api/src/test/java/org/dspace/administer/StructBuilderIT.java]: I've added a spot-check for a Collection handle and a Community with handle not specified.

Below the cited code, this test also applies the strategy from `testImportStructure`.

**Review by pnbecker — APPROVED** [review]
:+1: by code review
I must mention: a small part of that code goes back to a PR I created for @mwoodiupui (see https://github.com/mwoodiupui/DSpace/pull/13). So I'm kind of reviewing a small part of my own code. Even if it is a small PR, it should get a second review.

**Comment by mwoodiupui** [comment]
Yes, thanks to @pnbecker for the original code.  I chose to treat it as a separate issue rather than merge it into that other PR.

**Comment by vins01-4science** [comment]
> hi @mwoodiupui thanks for this PR and to ping us on the recent bug report about it! I made a quick code review and added inline a question for you about the test, can you take a look? my colleague will make a functional test in the next days as well

Hi @abollini, the functional tests are done, and I think we can accept these changes.
Thank you, @mwoodiupui.

**Comment by tdonohue** [comment]
Merging as this is at +2.  Thanks @mwoodiupui !

**Pull request merged**
